lib/auth/init_test.go

```diff
@@ -783,6 +783,65 @@ func TestInit_bootstrap(t *testing.T) {
 	}
 }
 
+const (
+	userYAML = `kind: user
+version: v2
+metadata:
+  name: joe
+spec:
+  roles: ["admin"]`
+	tokenYAML = `kind: token
+version: v2
+metadata:
+  name: github-token
+  expires: "3000-01-01T00:00:00Z"
+spec:
+  roles: [Bot]
+  join_method: github
+  bot_name: github-demo
+  github:
+    allow:
+      - repository: gravitational/example`
+)
+
+func TestInit_ApplyOnStartup(t *testing.T) {
+	t.Parallel()
+
+	user := resourceFromYAML(t, userYAML).(types.User)
+	token := resourceFromYAML(t, tokenYAML).(types.ProvisionToken)
+
+	tests := []struct {
+		name         string
+		modifyConfig func(*InitConfig)
+		wantErr      bool
+	}{
+		{
+			name: "Apply unsupported resource",
+			modifyConfig: func(cfg *InitConfig) {
+				cfg.ApplyOnStartupResources = append(cfg.ApplyOnStartupResources, user)
+			},
+			wantErr: true,
+		},
+		{
+			name: "Apply ProvisionToken",
+			modifyConfig: func(cfg *InitConfig) {
+				cfg.ApplyOnStartupResources = append(cfg.ApplyOnStartupResources, token)
+			},
+		},
+	}
+	for _, test := range tests {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			cfg := setupConfig(t)
+			test.modifyConfig(&cfg)
+
+			_, err := Init(cfg)
+			hasErr := err != nil
+			require.Equal(t, test.wantErr, hasErr, err)
```

Comment:
Let's put a `require.ErrorAssertionFunc` in the test case instead of a bool.

This way, if the test fails, we'll get an error message that is more helpful than "expected true, got false"